app/dashboard/models.py

```diff
@@ -4310,20 +4311,17 @@ def get_various_activities(self):
         """
 
         if not self.is_org:
-            activity_indexes = ActivityIndex.objects.filter(key=f'profile:{self.pk}').values_list('activity__pk', flat=True)
             all_activities = self.activities.all() | self.other_activities.all()
             return Activity.objects.filter(
-                pk__in=list(activity_indexes)
+                activities_index__key=f'profile:{self.pk}'
             ).order_by('-created').cache()
         else:
             # orgs
-            activity_indexes = ActivityIndex.objects.all().values_list('activity__pk', flat=True)
             url = self.github_url
             all_activities = Activity.objects.filter(
-                Q(pk__in=list(activity_indexes)) |
                 Q(bounty__github_url__istartswith=url) |
                 Q(tip__github_url__istartswith=url)
-            )
+            ).exclude(activities_index__key__isnull=True)
```

Comment:
Oh nice, this will clear everything from `Activity` if its not in the `ActivityIndex`?

Reply:
Yup yup I ran this on gitcoin4 to check